src/qpython.cpp

```diff
@@ -242,7 +242,18 @@ QPython::call(QVariant func, QVariant args, QJSValue callback)
     if (!callback.isNull() && !callback.isUndefined() && callback.isCallable()) {
         cb = new QJSValue(callback);
     }
-    emit process(func, args, cb);
+    // Unbox QJSValue from QVariant, since QJSValue::toVariant() can cause calls into
+    // QML engine and we don't want that to happen from non-GUI thread
+    QVariantList vl = args.toList();
+    for (int i = 0, c = vl.count(); i < c; ++i) {
```

Comment:
Use foreach here instead of loop variables, this should work:

```
foreach (QVariant &v, vl) {
    ...
}
```

As you are using a reference, doing `v = v.value<QJSValue>().toVariant();` should still update the value in the list (so you don't need `i` there for `vl[i]`).